src/Products/Products.Core/ValueObjects/AllergenCollection.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using IGroceryStore.Products.Entities;
+
+namespace IGroceryStore.Products.ValueObjects
+{
+    internal sealed class AllergenCollection : IEnumerable<Allergen>, IEquatable<AllergenCollection>
+    {
+        private ISet<Allergen> _allergens;
+        public static AllergenCollection Empty => new(Enumerable.Empty<Allergen>());
+
+        public IReadOnlySet<Allergen> Allergens
```

Comment:
maybe computed prop with ext method from this post?
https://stackoverflow.com/questions/36815062/c-sharp-hashsett-read-only-workaround